### PR TITLE
Further improvements in QDataGrid documentation.

### PR DESCRIPTION
**Pull request by vaibhav-kaushal**
Further improvements in QDataGrid documentation. QControlProxy documentation finished. Improved QColumn's property documentation. Other random improvements in syntax and documentation too.

### CODE REVIEW

File: includes/base_controls/QControlProxy.class.php [hunk 1]
```diff
@@ -90,7 +142,7 @@ public function __set($strName, $mixValue) {
 				
 				case 'TargetControlId': 
 					try {
-						return ($this->strFoo = QType::Cast($mixValue, QType::String));
```
Comment (olegabr) [inline includes/base_controls/QControlProxy.class.php]: OMG! why this `Foo` was here?

Comment (vaibhav-kaushal) [inline includes/base_controls/QControlProxy.class.php]: No idea mate! :stuck_out_tongue: I wanted to ask the same when I saw this. But there were so many things so I just corrected that and moved ahead.

**Comment by vaibhav-kaushal** [comment]
So, @olegabr and @spekary do you think documentation is coming out good? :blush:

**Comment by spekary** [comment]
Your efforts are very helpful. It looks like you have a particular style in mind. Do you want to create a style guide?

**Comment by vaibhav-kaushal** [comment]
@spekary I would like to but I hardly find time to document the project in first place. Once it is done, I have to resume writing the wiki content on qcubed.in. I had to document all functions one by one to talk about their effects and usage. With API documentation done, I could just redirect the function names (such as ```AddAction``` , which being a camel-cased word is automatically rendered as a link by the wiki's engine) to their corresponding API documentation link. 

Coming back to style: I use PhpStorm to extensively format the code. So I guess writing just that much would be enough? :wink:

**Comment by vaibhav-kaushal** [comment]
And I am getting the added benefit of exploring functions which are already there and sometimes doing _facepalm_ when i find that I spend a week designing a functionality which was already there!

**Pull request merged**